From 4fb6da70e12ae5dc1205ab40b3fdf135ce364ad0 Mon Sep 17 00:00:00 2001
From: Paul Thornthwaite <tokengeek@gmail.com>
Date: Tue, 19 Jun 2018 10:39:58 +0100
Subject: [PATCH] Make CloudSigma snapshot tests pending

There appears to be a number of bugs in the CloudSigma mock interface.

The `snapshots` key is `nil` and adding a new snapshot fails. Working
around that, the tests timeout because there is no mechanism to change
the snapshots state from "creating" so it never becomes `available?`
---
 tests/cloudsigma/models/snapshot_tests.rb    | 2 ++
 tests/cloudsigma/models/snapshots_tests.rb   | 1 +
 tests/cloudsigma/requests/snapshots_tests.rb | 1 +
 3 files changed, 4 insertions(+)

diff --git a/tests/cloudsigma/models/snapshot_tests.rb b/tests/cloudsigma/models/snapshot_tests.rb
index deba85c3e2..ec049fc151 100644
--- a/tests/cloudsigma/models/snapshot_tests.rb
+++ b/tests/cloudsigma/models/snapshot_tests.rb
@@ -1,4 +1,6 @@
 Shindo.tests('Fog::Compute[:cloudsigma] | snapshot model', ['cloudsigma']) do
+  pending if Fog.mocking?
+
   volume = Fog::Compute[:cloudsigma].volumes.create(:name => 'fogmodeltest', :size => 1024**3, :media => :disk)
   volume.wait_for { available? } unless Fog.mocking?
 
diff --git a/tests/cloudsigma/models/snapshots_tests.rb b/tests/cloudsigma/models/snapshots_tests.rb
index b42d037c65..2aedf4f32b 100644
--- a/tests/cloudsigma/models/snapshots_tests.rb
+++ b/tests/cloudsigma/models/snapshots_tests.rb
@@ -1,4 +1,5 @@
 Shindo.tests('Fog::Compute[:cloudsigma] | snapshots collection', ['cloudsigma']) do
+  pending if Fog.mocking?
 
   volume = Fog::Compute[:cloudsigma].volumes.create(:name => 'fogtest', :size => 1024**3, :media => :disk)
   volume.wait_for { available? } unless Fog.mocking?
diff --git a/tests/cloudsigma/requests/snapshots_tests.rb b/tests/cloudsigma/requests/snapshots_tests.rb
index c58cb381ea..f2b240919a 100644
--- a/tests/cloudsigma/requests/snapshots_tests.rb
+++ b/tests/cloudsigma/requests/snapshots_tests.rb
@@ -1,4 +1,5 @@
 Shindo.tests('Fog::Compute[:cloudsigma] | snapshot requests', ['cloudsigma']) do
+  pending if Fog.mocking?
 
   @snapshot_format = {
       'uuid' => String,
